platform/darwin/src/MGLGeometry.h

```diff
@@ -45,6 +45,23 @@ typedef struct MGLCoordinateBounds {
     CLLocationCoordinate2D ne;
 } MGLCoordinateBounds;
 
+/** 
+ A quadrilateral area as measured on a two-dimensional map projection.
+ `MGLCoordinateQuad` differs from `MGLCoordinateBounds` in that it allows
+ representation of non-axis aligned bounds and non-rectangular quadrilaterals
+ */
+typedef struct MGLCoordinateQuad {
+    /** Coordinate at the top left corner. */
+    CLLocationCoordinate2D topLeft;
+    /** Coordinate at the top right corner. */
+    CLLocationCoordinate2D topRight;
+    /** Coordinate at the bottom right corner. */
+    CLLocationCoordinate2D bottomRight;
+    /** Coordinate at the bottom left corner. */
+    CLLocationCoordinate2D bottomLeft;
```

Comment:
`NSEdgeInsets` and `UIEdgeInsets` fields have a different intent than the `MGLCoordinateQuad`. The insets are referring to relative distances between rects, in what is the common order to think about rect fields - top, left, bottom, right. I don't think that they really qualify as counterclockwise.
The quad has two intents - specifying the orientation the image on screen, and the non-axis aligned bounds of the image. Images are usually looked at top-down, and it makes sense to have the coordinates be provided in clockwise order.

To your point though, the geojson spec for [Polygons](https://tools.ietf.org/html/rfc7946#section-3.1.6) expects outer rings to be in counter clockwise order.

If we were providing pure geometric information, I would agree to using a counterclockwise order. Given that it also provides orientation, I am torn on what is _better_